components/aws/sagemaker/train/src/train.py

```diff
@@ -75,6 +78,7 @@ def signal_term_handler(signalNumber, frame):
   logging.info('Job request submitted. Waiting for completion...')
   try:
     _utils.wait_for_training_job(client, job_name)
+    _utils.wait_for_debug_rules(client, job_name)
```

Comment:
We recently added the ability to stop training job when user initiates a terminate run on a InProgress training job. 

With debugger there is a case where training job is finished but kfp pipeline is still in progress waiting for debugger processing to finish, if user clicks a terminate at this point, the background debugging/processing jobs should be stopped